### PR TITLE
Fixes #3608 - Use --user flag to install locally for the current user

### LINKED ISSUE #3608: Unnecessary "becomes" pollutes global python install
## Environment details:

* build system

## Steps to reproduce the issue:

1.   Try and use the ansible scripts in a build env that doesn't do things globally
2.   The ansible install for pip affects the global packages on the system

## Provide the expected results and outputs:

The prereqs steps install versions of specific python packages via sudo this isn't necessary.

## Provide the actual results and outputs:

The packages should be installed locally just to the user running the build. That way it pollutes the environment a bit less.

## Additional information you deem important:
* This doesn't break travis and makes it easier for people to use the build in a local jenkins.

### PR DESCRIPTION
**Pull request by mcdan**
Closes #3610 

## Related issue and scope

- [x] I opened an issue to propose and discuss this change (#????)

## My changes affect the following components

- [ ] API
- [ ] Controller
- [ ] Message Bus (e.g., Kafka)
- [ ] Loadbalancer
- [ ] Invoker
- [ ] Intrinsic actions (e.g., sequences, conductors)
- [ ] Data stores (e.g., CouchDB)
- [ ] Tests
- [ ] Deployment
- [ ] CLI
- [x] General tooling
- [ ] Documentation

## Types of changes

- [x] Bug fix (generally a non-breaking change which closes an issue).
- [ ] Enhancement or new feature (adds new functionality).
- [ ] Breaking change (a bug fix or enhancement which changes existing behavior).

## Checklist:

- [x] I signed an [Apache CLA](https://github.com/apache/incubator-openwhisk/blob/master/CONTRIBUTING.md).
- [x] I reviewed the [style guides](https://github.com/apache/incubator-openwhisk/wiki/Contributing:-Git-guidelines#code-readiness) and followed the recommendations (Travis CI will check :).
- [x] I added tests to cover my changes.
- [x] My changes require further changes to the documentation.
- [x] I updated the documentation where necessary.

### CODE REVIEW

File: ansible/roles/prereq/tasks/deploy.yml [hunk 1]
```diff
@@ -30,16 +30,16 @@
   pip:
     name: requests
     version: 2.10.0
-  become: true
+    extra_args: --user
 
 - name: install docker for python
-  pip:
+  pip:
```
Comment (rabbah) [inline ansible/roles/prereq/tasks/deploy.yml]: trailing white space

**Comment by rabbah** [comment]
@csantanapr probably want a PG for this to be sage.

**Comment by mcdan** [comment]
Yes: https://github.com/apache/incubator-openwhisk/issues/3608
This is causing issues on a jenkins pipeline build I'm working on. Our jenkins users don't have access to install stuff globally.

**Comment by csantanapr** [comment]
The way I did for the other tasks is using “become: <book var>”

Using become: false works for you instead ?

**Comment by rabbah** [comment]
@mcdan there are a lot of `become: true`s in the scripts --- are you sure this is the only place you need this?

i know @csantanapr and I tried to remove the privilege escalation before but not finding it easily. I posit it should not be necessary.

**Comment by mcdan** [comment]
@rabbah We are setting up a new jenkins so I will get back to this next week, I will verify if we need any more and report back.

**Comment by rabbah** [comment]
@csantanapr ok to merge?

**Comment by dgrove-oss** [comment]
@csantanapr  - Should this be closed; sounds like the approach isn't viable?

**Pull request closed without merging**